### PR TITLE
Fix being able to have replacement variables substituted into email templates

### PR DESCRIPTION
**Pull request by branson-dyer**

### CODE REVIEW

**Review by domdomegg — COMMENTED** [review]
Awesome work, thank you for contributing a fix. There's a couple minor comments to address and I can get this merged and released.

This page might explain my review comments a little more: https://adamjones.me/reviews/ :)

File: src/store.ts [hunk 1]
```diff
@@ -56,7 +61,7 @@ export const clearStore = () => {
 // This type doesn't give us perfect readonly safety
 // But this is probably safe enough for now, given the method name
 // and the relatively small project size.
-export const getStoreReadonly = (): Readonly<Store> => store;
+export const getStoreReadonly = (): Readonly<StoreReadOnly> => ({emails:store.emails, templates:Object.fromEntries(store.templates)});
```
[the remark this answers is not in the document]
Comment (branson-dyer) [reply src/store.ts:64]: Yeah you are correct on that 
image

File: src/store.ts [hunk 2]
```diff
@@ -6,6 +6,11 @@ export interface Store {
   templates: Map<string, Template>,
 }
 
+export interface StoreReadOnly {
```
[the remark this answers is not in the document]
Comment (branson-dyer) [reply src/store.ts:9]: The intention here was to separate the map and object versions, as the map version doesn't get sent to the frontend by express. I'll look into that as well, I haven't used typescript before so this was just an easy choice for me.

File: src/v2/sendBulkEmail.ts [hunk 3]
```diff
@@ -137,9 +139,15 @@ const handleBulk: RequestHandler = async (req, res) => {
 /**
  * Decode template data.
  */
-function decodeTemplateData(templateData?: string): Replacement[] {
-  const result = templateData ? JSON.parse(templateData) : [];
-  return result ?? [];
+function decodeTemplateData(templateData?: string): [Replacement[], string] {
```
[the remark this answers is not in the document]
Comment (branson-dyer) [reply src/v2/sendBulkEmail.ts:142]: Yeah I was just doing this how I would in golang, will implement that

File: src/store.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (branson-dyer) [reply src/store.ts]: For this one I just moved this over to the /store res.send. If you want me to make it readonly I can

File: src/store.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (branson-dyer) [reply src/store.ts]: As above I reverted this change moving the map->object convertion to res.send

**Review by domdomegg — APPROVED** [review]
Looks great, thank you! Will get this released shortly.

**Comment by domdomegg** [comment]
Released in 2.4.1, now published to npm

**Pull request merged**